CrewChiefV4/Events/LapTimes.cs

```diff
@@ -363,32 +359,58 @@ protected override void triggerInternal(GameStateData previousGameState, GameSta
                     (currentGameState.OpponentData.Count == 0 || (currentGameState.OpponentData.Count == 1 && currentGameState.OpponentData.First().Value.DriverRawName == currentGameState.SessionData.DriverRawName));
                 if (isHotLappingOrLonePractice)
                 {
+                    // note that lone practice in changing conditions doesn't take conditions into account. This is a bit of an edge case
                     lapAndSectorsComparisonData[0] = lapAndSectorsSelfComparisonData[0];
                     lapAndSectorsComparisonData[1] = lapAndSectorsSelfComparisonData[1];
                     lapAndSectorsComparisonData[2] = lapAndSectorsSelfComparisonData[2];
                     lapAndSectorsComparisonData[3] = lapAndSectorsSelfComparisonData[3];
                 }
                 else
                 {
+                    // in qual sessions we want absolute timings. We can also use absolute timings if the conditions are static.
+                    // If the conditions are changing we want timings relative to the prevailing conditions for non-qual sessions.
+                    // For race sessions we want the recent pace
                     if (currentGameState.SessionData.SessionType == SessionType.Race)
                     {
-                        lapAndSectorsComparisonData = currentGameState.getTimeAndSectorsForBestOpponentLapInWindow(paceCheckLapsWindowForRaceToUse, currentGameState.carClass);
-                        Console.WriteLine("Opponents best sectors = " + lapAndSectorsComparisonData[1].ToString("0.000") + ",    " + lapAndSectorsComparisonData[2].ToString("0.000") + ",    " + lapAndSectorsComparisonData[3].ToString("0.000"));
-                        Console.WriteLine("Player best sectors = " + currentGameState.SessionData.PlayerBestSector1Time.ToString("0.000") + ",    " + currentGameState.SessionData.PlayerBestSector2Time.ToString("0.000") + ",    " + currentGameState.SessionData.PlayerBestSector3Time.ToString("0.000"));
+                        if (!currentGameState.TimingData.conditionsHaveChanged)
```

Comment:
I think it's better to hide this variability (conditionsHaveChanged) inside of TimingData methods.

Reply:
while i certainly agree with this, it's part of a (much) larger refactor where all the timing data is moved to the TimingData class. I made a start on this refactor this morning and it's a huge and complex job - it's also very invasive. I think it needs doing but not until i've got 2 or 3 clear days to focus on it. In the mean time i don't want to block possible releases